code/datums/components/dusting.dm

```diff
@@ -0,0 +1,249 @@
+/datum/component/dusting
+	/// Consume all objects that bump into the atom. Only applies to dense atoms.
+	var/consume_on_bumped = TRUE
+	/// Consume when attacking with the sword
+	var/consume_on_attack = TRUE
+	/// Consume when attacked by a mob with attack_hand
+	var/consume_on_attack_hand = TRUE
+	/// Consume when attacked by a mob using an item
+	var/consume_on_attackby = TRUE
+	/// Consume when trying to get picked up by a mob
+	var/consume_on_pickup = FALSE
+	/// Consume when we are thrown onto something
+	var/consume_on_throw = TRUE
+	/// If we consume the contents of atoms we consume as well. This ignores turfs regardless if consume_turfs is TRUE
+	var/consume_all_contents = FALSE
+	/// If we consume turfs
+	var/consume_turfs = FALSE
+	/// The sound to play when consuming something
+	var/consumption_sound = 'sound/effects/supermatter.ogg'
+	/// Radiation arguments for when we consume a specific atom
+	var/list/radiation_atom = list(
+		"default" = list("range"=6,"threshold"=0.6,"chance"=60),
+		/obj = list("range"=3,"threshold"=0.1,"chance"=30),
+	)
+	/// Range to send consumption texts
+	var/consume_text_range = 10
+	/// Text to show to deaf people when something had been consumed. %parent% for parent this component is attached to, %target% to what got consumed
+	var/consume_text_visual = span_danger("As %parent% slowly stops resonating, you find your skin covered in new radiation burns.")
+	/// Audible text to show to people when something had been consumed. %parent% for parent this component is attached to, %target% to what got consumed
+	var/consume_text_audible = span_danger("The unearthly ringing subsides and you find your skin covered in new radiation burns.")
+	/// Audible text to show to people when they can't see what's being consumed, but are still in range to recieve consumtion messages. %parent% for parent this component is attached to, %target% to what got consumed
+	var/consume_text_unseen = span_hear("An unearthly ringing fills your ears, and you find your skin covered in new radiation burns.")
+	/// If a target has this flag, don't consume it
+	var/ignore_flags = SUPERMATTER_IGNORES_1
+	/// If a mob has this flag, don't consume it, along with ignore_flags
+	var/ignore_status_flags = GODMODE
+	/// Ignore subtypes of this type to consume. Becomes typecache list
+	var/list/ignore_subtypesof = list(/obj/effect, /obj/item/scalpel/supermatter)
+
+	/// Currently not sending any messages to admins to prevent chat spam. Not a bool
```

Comment:
Well what is it then?

Reply:
Its a var that ranges from 0 to 2 cause I'm too lazy for defines. It's mostly for logging when the component is allowed to consume all contents of a target and doesn't flood chat

Reply:
then write that in the documentation, ya?